activerecord/test/cases/fixtures_test.rb

```diff
@@ -310,6 +310,10 @@ def test_has_many_through_with_renamed_table
     assert_equal load_has_and_belongs_to_many['parrots_treasures'], rows['parrot_treasures']
   end
 
+  def test_has_and_belongs_to_many_order
+    assert_equal ['parrots', 'parrots_treasures'], load_has_and_belongs_to_many.keys
```

Comment:
This is testing an implementation detail, not the effect. Let's just try loading a HABTM with a foreign key and test that it works.